File: 1993-SumOfAllSubsetXorTotals/1993-SumOfAllSubsetXorTotals.java
// Last updated: 6/4/2026, 10:26:13 AM
class Solution {
    public int subsetXORSum(int[] nums) {
        int res=0;
        int n = nums.length;
        for(int i=0;i<n;i++){
            res = res|nums[i];
        }
        return res<<(n-1);
    }
}

File: 1993-SumOfAllSubsetXorTotals/1993-SumOfAllSubsetXorTotals.cpp
// Last updated: 6/4/2026, 10:30:16 AM
class Solution {
public:
int n;
    int f(int i, int xorsum, vector<int>& nums){
        if (i==n) return xorsum;
        int x=nums[i];
        int take=f(i+1, xorsum^x, nums);
        int notake=f(i+1, xorsum, nums);
        return take+notake;
    }
    int subsetXORSum(vector<int>& nums) {
        n=nums.size();
        return f(0, 0, nums);
    }
};